cmd/stats/stats.go

```diff
@@ -2,28 +2,223 @@ package stats
 
 import (
 	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"strings"
+
+	. "github.com/NicksPatties/sweet/db"
 	"github.com/spf13/cobra"
 )
 
-var Command = &cobra.Command{
+var Cmd = &cobra.Command{
 	Use:   "stats",
 	Short: "Print statistics about typing exercises",
 	Run: func(cmd *cobra.Command, args []string) {
-		since, _ := cmd.Flags().GetString("since")
-		printStats(since)
+		q, err := argsToQuery(cmd, time.Now())
+		if err != nil {
+			log.Fatal()
+			os.Exit(-1)
+		}
+		reps, err := queryToReps(q)
+		if err != nil {
+			log.Fatal()
+			os.Exit(-1)
+		}
+		cols := argsToColumnFilter(cmd)
+		printStats(reps, cols)
 	},
 }
 
-func printStats(since string) {
-	if since != "" {
-		fmt.Printf("Printing stats since %s\n", since)
+// Returns the correct date depending on the start argument.
+// The time returned is at midnight of the date specified by `start`,
+// unless an hour query is used, in which case it will return the time
+// specified by
+func parseDateFromArg(isStart bool, arg string, now time.Time) (time.Time, error) {
+	var todayAtMidnight time.Time
+	todayAtMidnight = time.Date(now.Year(), now.Month(), now.Day(),
+		0, 0, 0, 0, now.Location())
+
+	if !isStart {
```

Comment:
Would it be worth it to be more explicity in what `!isStart` means?

```go
if isEnd := !isStart; isEnd {
  // ...
}
```

Reply:
It would be better to modify the arguments to be `isEnd`, but this means I need to modify all my tests to reflect that.